### ISSUE TITLE
MA0038 should not be raised when its fix leads to MA0018

### ISSUE THREAD

**Issue by nj0yeh** [meziantou/Meziantou.Analyzer#148]
MA0038 should not be raised when its fix leads to MA0018

```
public abstract class TestBase<T> : OtherBase<dynamic, T>
  where T : OtherBase, new()
{
    protected void NiceMethod()
    {
        Console.WriteLine("Hello world!");
    }
}
```

**Comment by meziantou** [comment]
MA0018 was intended to report diagnostic on public method. So, it was not expected to report in this case. This is fixed in the latest version (1.0.399)

That's being said, I think an analyzer should report a diagnostic even if the fix leads to another rule. It would be better if the initial fix could fix all rules at once but some rules cannot be fixed automatically as for MA0018. Moreover, I don't think I can detect which rules are active in the solution. So, if a user disable the other rule, I would not report the first one even if the fix would not lead to raise another diagnostic.

Note that if you don't want to fix a diagnostic, you can suppress it by using an attribute (`[SuppressMessage]`). If a rule raises lots of false positives, I agree the analyzer should be fixed. But I don't think this is the case here.

**Comment by nj0yeh** [comment]
Thank you for the quick fix and the explanation! I agree.
